src/common/dto/view/CreateView.dto.ts

```diff
@@ -0,0 +1,30 @@
+import { ApiProperty } from '@nestjs/swagger';
+import {
+  IsEnum,
+  IsString,
+  MinLength,
+} from 'class-validator';
+import { COURSE_TABLE_COLUMN } from 'common/constants';
+
+/**
+ * Create View DTO
+ *
+ * Defines the data shape for creating a new [[View]] entry in the database.
+ */
+export class CreateViewDTO {
+  @ApiProperty({
+    description: 'Array of column names for this view',
+    example: COURSE_TABLE_COLUMN.AREA,
+    enum: COURSE_TABLE_COLUMN,
+  })
+  @IsEnum(COURSE_TABLE_COLUMN, { each: true })
+  public columns: COURSE_TABLE_COLUMN[] = [];
```

Comment:
I was going to suggest making sure the array is not empty, but I suppose it should be okay for someone to create a view with no columns at first if they wanted. What do you think?

Reply:
Yeah, as much as it doesn't make sense logically - I think we might want to allow this.